### PR TITLE
Better ID choice for new item.

### PR DESCRIPTION
**Pull request by vrabaud**
This fixes the case where the maximum ID is UINT32_MAX.

BUG=oss-fuzz:65657

### CODE REVIEW

File: src/read.c [hunk 1]
```diff
@@ -4357,7 +4353,26 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
...
+        for (uint32_t i = 0; i < meta->items.count; ++i) {
+            if (meta->items.item[i]->id == newItemID) {
+                isUsed = AVIF_TRUE;
+                break;
+            }
+        }
+    } while (isUsed && newItemID != 0);
+    avifResult result;
+    if (newItemID == 0) {
+        // In the improbable case where all IDs are used.
```
Comment (y-guyon) [inline src/read.c:4371]: Some test such as the following might be clearer than implicit uint32_t looping overflow:

```c
AVIF_CHECKERR(meta->items.count < MAX_UINT32_T - 1, AVIF_RESULT_DECODE_ALPHA_FAILED);
```

**Review by y-guyon — APPROVED** [review]

File: src/read.c (hunk 1 above)
Comment (vrabaud) [reply src/read.c:4360]: Done

File: src/read.c [hunk 2]
```diff
@@ -4357,7 +4354,26 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
...
+        do {
+            ++newItemID;
+            isUsed = AVIF_FALSE;
+            for (uint32_t i = 0; i < meta->items.count; ++i) {
+                if (meta->items.item[i]->id == newItemID) {
+                    isUsed = AVIF_TRUE;
+                    break;
+                }
+            }
+        } while (isUsed && newItemID != 0);
```
[the remark this answers is not in the document]
Comment (vrabaud) [reply src/read.c:4374]: Right, algorithmically, you are correct, but it is typically the kind of case where I do not want to make an infinite loop possible. I'd rather have the assert fail later in avifMetaFindOrCreateItem than loop forever here.
You are right, it cannot happen right now. But what if somebody switches it to a uint16_t tomorrow? It is a stretch but I'd rather be safe.

[the remark this answers is not in the document]
Comment (wantehchang) [reply src/read.c:4374]: Thanks for confirming you intended to keep the `newItemID != 0` check. (I was wondering if it was an oversight.)

We should handle `newItemID == 0` after the do-while loop. We should not rely on assertions for error handling.

[the remark this answers is not in the document]
Comment (vrabaud) [reply src/read.c:4374]: Indeed, done in https://github.com/AOMediaCodec/libavif/pull/1967

**Pull request merged**